File: frontend/src/app/core/services/analytics.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import { API_ENDPOINTS } from '../constants/api.constants';
import { AnalyticsSummary } from '../models/analytics.models';
import { JobViewRequest } from '../models/notification.models';

@Injectable({ providedIn: 'root' })
export class AnalyticsService {
  private readonly http = inject(HttpClient);

  recordJobView(jobId: number, payload: JobViewRequest = {}): Observable<number> {
    return this.http.post<number>(`${API_ENDPOINTS.analytics}/jobs/${jobId}/views`, payload);
  }

  getJobViewCount(jobId: number): Observable<number> {
    return this.http.get<number>(`${API_ENDPOINTS.analytics}/jobs/${jobId}/view-count`);
  }

  getApplicationCount(jobId: number): Observable<number> {
    return this.http.get<number>(`${API_ENDPOINTS.analytics}/jobs/${jobId}/application-count`);
  }

  getViewToApplyRatio(jobId: number): Observable<number> {
    return this.http.get<number>(`${API_ENDPOINTS.analytics}/jobs/${jobId}/view-to-apply-ratio`);
  }

  getRecruiterStats(recruiterId: number): Observable<AnalyticsSummary> {
    return this.http.get<AnalyticsSummary>(`${API_ENDPOINTS.analytics}/recruiter/${recruiterId}`);
  }

  getTimeToHire(recruiterId: number): Observable<number> {
    return this.http.get<number>(`${API_ENDPOINTS.analytics}/recruiter/${recruiterId}/time-to-hire`);
  }

  getPlatformStats(): Observable<AnalyticsSummary> {
    return this.http.get<AnalyticsSummary>(`${API_ENDPOINTS.analytics}/admin`);
  }

  getTopCategories(): Observable<Record<string, number>> {
    return this.http.get<Record<string, number>>(`${API_ENDPOINTS.analytics}/categories/top`);
  }
}

File: hireconnect-web/src/main/java/com/hireconnect/web/service/AnalyticsService.java
package com.hireconnect.web.service;

import java.util.Map;

import org.springframework.core.ParameterizedTypeReference;
import org.springframework.stereotype.Service;

import com.hireconnect.web.dto.AnalyticsSummary;
import com.hireconnect.web.dto.PortalSession;
import com.hireconnect.web.support.GatewayClient;

@Service
public class AnalyticsService {

    private final GatewayClient gatewayClient;

    public AnalyticsService(GatewayClient gatewayClient) {
        this.gatewayClient = gatewayClient;
    }

    public AnalyticsSummary getRecruiterStats(Integer recruiterId, PortalSession session) {
        return gatewayClient.get("/api/v1/analytics/recruiter/{recruiterId}", session, AnalyticsSummary.class, recruiterId);
    }

    public Double getTimeToHire(Integer recruiterId, PortalSession session) {
        return gatewayClient.get("/api/v1/analytics/recruiter/{recruiterId}/time-to-hire", session, Double.class, recruiterId);
    }

    public AnalyticsSummary getPlatformStats(PortalSession session) {
        return gatewayClient.get("/api/v1/analytics/admin", session, AnalyticsSummary.class);
    }

    public Integer getJobViewCount(Integer jobId, PortalSession session) {
        return gatewayClient.get("/api/v1/analytics/jobs/{jobId}/view-count", session, Integer.class, jobId);
    }

    public Integer getAppCountByJob(Integer jobId, PortalSession session) {
        return gatewayClient.get("/api/v1/analytics/jobs/{jobId}/application-count", session, Integer.class, jobId);
    }

    public Double getViewToApplyRatio(Integer jobId, PortalSession session) {
        return gatewayClient.get("/api/v1/analytics/jobs/{jobId}/view-to-apply-ratio", session, Double.class, jobId);
    }

    public Map<String, Long> getTopJobCategories(PortalSession session) {
        return gatewayClient.get("/api/v1/analytics/categories/top", session, new ParameterizedTypeReference<>() {
        });
    }
}
